### PR TITLE
Tm4 c1294 can char dev

### PR DESCRIPTION
**Pull request by dvosahlik**
## Summary
Implementation of TI Tiva TM4C1294 launchpad character device driver. The driver was taken over from the 1293 launchpad. Minor modifications were added to the configuration file. Bugfix of typo in the arch/tiva/tiva_can.c file
## Impact
Added support for missing CAN implementation on the TM4C1294XL launchpad
## Testing
The implementation was tested using TM4C1294XL launchpad against the PEAK USB CAN converter using can0 device on the lauchpad.

### CODE REVIEW

File: boards/arm/tiva/tm4c1294-launchpad/src/tm4c_can.c [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/****************************************************************************
+ * boards/arm/tiva/tm4c1294-launchpad/src/tm4c_can.c
+ * Based heavily on stm32_can.c from the boards directory.
+ *
+ * Redistribution and use in source and binary forms, with or without
```
Comment (xiaoxiang781216) [inline boards/arm/tiva/tm4c1294-launchpad/src/tm4c_can.c:5]: need use apache copyright for new file

**Review by acassis — APPROVED** [review]

File: boards/arm/tiva/tm4c1294-launchpad/Kconfig [hunk 2]
```diff
@@ -72,4 +72,46 @@ config TM4C1294_LAUNCHPAD_TIMER_ALTCLK
 
 endif # TM4C1294_LAUNCHPAD_TIMER
 endif # TIVA_TIMER32_PERIODIC
+config TM4C1294_LAUNCHPAD_JUMPERS_CAN
+	bool "Use CAN jumpers configuration"
+	default n
+	select TIVA_UART2
+	---help---
+		If set, Nuttx will assume the jumpers JP4 and JP5 are set in CAN configuration.
```
Comment (acassis) [inline boards/arm/tiva/tm4c1294-launchpad/Kconfig:80]:
```suggestion
		If set, NuttX will assume the jumpers JP4 and JP5 are set in CAN configuration.

File: boards/arm/tiva/tm4c1294-launchpad/src/tm4c1294-launchpad.h [hunk 3]
```diff
@@ -77,6 +77,9 @@
 #  undef HAVE_I2CTOOL
 #endif
 
+/* Define the procfs mounting point */
+#define TIVA_PROCFS_MOUNTPOINT "/proc"
```
Comment (acassis) [inline boards/arm/tiva/tm4c1294-launchpad/src/tm4c1294-launchpad.h:81]: Please add empty line after comment line

Comment (dvosahlik) [reply boards/arm/tiva/tm4c1294-launchpad/src/tm4c1294-launchpad.h:82]: Done in [997e60a](https://github.com/apache/nuttx/pull/8152/commits/997e60aaddec0694f4ca171eeeae0adc57b66ab3)

File: boards/arm/tiva/tm4c1294-launchpad/Kconfig (hunk 2 above)
Comment (dvosahlik) [reply boards/arm/tiva/tm4c1294-launchpad/Kconfig]: Done in [997e60a](https://github.com/apache/nuttx/pull/8152/commits/997e60aaddec0694f4ca171eeeae0adc57b66ab3)

File: boards/arm/tiva/tm4c1294-launchpad/src/tm4c_can.c (hunk 1 above)
Comment (dvosahlik) [reply boards/arm/tiva/tm4c1294-launchpad/src/tm4c_can.c]: Done in [997e60a](https://github.com/apache/nuttx/pull/8152/commits/997e60aaddec0694f4ca171eeeae0adc57b66ab3)

**Comment by xiaoxiang781216** [comment]
@dvosahlik please squash the last patch to the second one.

**Comment by dvosahlik** [comment]
@xiaoxiang781216, I did the squash as requested.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**